@uportal/waffle-menu/src/components/WaffleMenu.vue

```diff
@@ -0,0 +1,341 @@
+<template>
+  <div
+    class="waffle-menu-container"
+    ref="waffleMenu"
+  >
+    <button
+      class="waffle-trigger"
+      @click="toggleMenu()"
+    >
+      <FontAwesomeIcon
+        icon="th"
+        :color="buttonColor"
+        size="2x"
+      />
+      <div v-if="menuOpen">
+        <div class="waffle-triangle-black" />
+        <div
+          class="waffle-triangle"
+          :style="{
+            borderColor:
+              'transparent transparent ' + menuBackgroundColor + ' transparent',
+          }"
+        />
+      </div>
+    </button>
+    <ul
+      class="waffle-dropdown"
+      v-if="menuOpen && dataLoaded"
+      :style="{ backgroundColor: menuBackgroundColor }"
+    >
+      <li
+        class="waffle-menu-item"
+        v-for="(item, index) in this.dataMenuItems"
+        :key="index"
+      >
+        <a
+          :href="item.link"
+          :background="item.image"
+        >
+          <img
+            :src="item.image"
+            alt=""
+          > <span>{{ item.label }}</span>
+        </a>
+      </li>
+      <li
+        class="waffle-dropdown-footer"
+        v-for="(item, index) in this.dataMenuFooter"
+        :key="index"
+      >
+        <a :href="item.link">
+          {{ item.label }}
+        </a>
+      </li>
+    </ul>
+  </div>
+</template>
+<script>
+import oidc from '@uportal/open-id-connect';
+import {portletRegistryToArray} from '@uportal/portlet-registry-to-array';
+import get from 'lodash/get';
+// import PropTypes from 'prop-types';
+import {library} from '@fortawesome/fontawesome-svg-core';
+import {faTh} from '@fortawesome/free-solid-svg-icons';
+import {FontAwesomeIcon} from '@fortawesome/vue-fontawesome';
+
+library.add(faTh);
+
+export default {
+  name: 'WaffleMenu',
+  components: {
+    FontAwesomeIcon,
+  },
+  props: {
+    url: {
+      type: String,
+      default: '/uPortal/api/v4-3/dlm/portletRegistry.json',
+    },
+    oidcUrl: {
+      type: String,
+      default: '/uPortal/api/v5-1/userinfo',
+    },
+    buttonColor: {
+      type: String,
+      default: '#fff',
+    },
+    menuBackgroundColor: {
+      type: String,
+      default: '#fff',
+    },
+    debug: {
+      type: Boolean,
+      default: false,
+    },
+    defaultIcon: {
+      type: String,
+      default:
+        '/ResourceServingWebapp/rs/tango/0.8.90/32x32/categories/applications-other.png',
+    },
+    truncateLength: {
+      type: Number,
+      default: 50,
+    },
+  },
+
+  computed: {
+    dataMenuItems: function() {
+      return this.data.filter((datum) => datum.type === 'box');
+    },
+    dataMenuFooter: function() {
+      return this.data.filter((datum) => datum.type === 'footer');
```

Comment:
I'm not sure I understand what's going on here (though I can see this stuff was carried over from the pre-Vue component).  Where is an item with `type=footer` going to come from?  (I don't know of any such thing.)

Reply:
This could probably be a slot in the Vue component

Reply:
It is my understanding that that would be the "show all" link that would appear at the bottom of the menu.  I assumed that it was part of the data returned from the API call.  Is that not the case?

Reply:
There's no data like that from the API.

Folks -- let's punt this one to the next update/pr to this component.  We need to cross some sort of finish line.

Reply:
created #261